server/main.lua

```diff
@@ -59,36 +59,64 @@ end)
 ----   Functions   ----
 -----------------------
 
+--- Gives the user the keys to the vehicle
+--- @param source number ID of the player
+--- @param plate string The plate number of the vehicle.
 function GiveKeys(source, plate)
+    local citizenid = getCitizenId(source)
+
+    if not citizenid then return end
+
     local keys = Player(source).state.keysList or {}
 
     if keys[plate] then return end
     keys[plate] = true
-    Player(source).state:set('keysList', keys, true)
-    local citizenid = getCitizenId(source)
 
-    if not citizenid then return end
+    Player(source).state:set('keysList', keys, true)
 
     if not keysList[citizenid] then
-        keysList[citizenid] = {}
+        keysList[citizenid] = {plate = true}
+    else
+        keysList[citizenid][plate] = true
     end
 
-    keysList[citizenid][plate] = true
-
     exports.qbx_core:Notify(source, locale('notify.keys_taken'))
 end
 
+RegisterNetEvent('qbx_vehiclekeys:server:giveKeys', function(source, plate)
+    if not GetInvokingResource() then return end
+    GiveKeys(source, plate)
+end)
+
 exports('GiveKeys', GiveKeys)
 
+--- Removing the vehicle keys from the user
+--- @param source number ID of the player
+--- @param plate string The plate number of the vehicle.
```

Comment:
By convention we normally don't include a space between the dash and the annotation.
```suggestion
---@param source number ID of the player
---@param plate string The plate number of the vehicle.
```

Reply:
how about a summary?